### PR TITLE
Fix the script `check_make_sizegen`

### PR DESCRIPTION
**Pull request by frouioui**
## Description

This PR fixes and simplifies the `check_make_sizegen` script. It was previously failing as we were moving all the `cache_file` to the `/tmp` folder. With this change, the script has a similar structure as the `check_make_proto` script and runs correctly.

The failures could be observed in the CI, although the build was still successful.

## Checklist

-   [x] "Backport me!" label has been added if this change should be backported
-   [x] Tests were added or are not required
-   [x] Documentation was added or is not required

### CODE REVIEW

File: tools/check_make_sizegen.sh [hunk 1]
```diff
@@ -1,49 +1,22 @@
...
-  fi
-  mv "$TMP" "$SRC"
-done
+second_output=$(git status --porcelain)
 
-exit $STATUS
+diff=$(diff <( echo "$first_output") <( echo "$second_output"))
 
+if [[ "$diff" != "" ]]; then
+  echo "ERROR: Regenerated cache_size files do not match the current version."
```
Comment (deepthi) [inline tools/check_make_sizegen.sh:19]:
```suggestion
  echo "ERROR: Regenerated cached_size files do not match the current version."
```

**Review by deepthi — COMMENTED** [review]
👍 
Can we collect status from each check and fail at the end if one or more have failed?
cc @rohit-nayak-ps

**Comment by frouioui** [comment]
> Can we collect status from each check and fail at the end if one or more have failed?

@deepthi, we can and we should. I'll update this PR.

File: tools/check_make_sizegen.sh [hunk 2]
```diff
@@ -1,49 +1,22 @@
 #!/bin/bash
+
 #
 # Validate that the current version of the generated cache_size files match the output
 # generated by sizegen.
 #
-# This is used in Travis to verify that the currently committed version was
```
Comment (ajm188) [inline tools/check_make_sizegen.sh:6]: > in Travis

lol

**Review by ajm188 — APPROVED** [review]
LGTM

**Pull request merged**